xds/src/test/java/io/grpc/xds/RingHashLoadBalancerTest.java

```diff
@@ -150,7 +150,8 @@ public void subchannelLazyConnectUntilPicked() {
     assertThat(result.getStatus().isOk()).isTrue();
     assertThat(result.getSubchannel()).isNull();
     Subchannel subchannel = Iterables.getOnlyElement(subchannels.values());
-    int expectedTimes = PickFirstLoadBalancerProvider.isEnabledHappyEyeballs() ? 1 : 2;
+    int expectedTimes = PickFirstLoadBalancerProvider.isEnabledHappyEyeballs()
+                            || !PickFirstLoadBalancerProvider.isEnabledNewPickFirst() ? 2 : 1;
```

Comment:
It doesn't seem the `isEnabledHappyEyeballs()` is really doing anything. Should this just be (aka, adjusting the original code to be s/isEnabledHappyEyeballs/isEnabledNewPickFirst/):
```java
PickFirstLoadBalancerProvider.isEnabledNewPickFirst() ? 1 : 2
```